dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/PreCheckSecurityOwnershipValidationHook.java

```diff
@@ -206,10 +201,42 @@ public void validateEnrollment( ValidationErrorReporter reporter, Enrollment enr
             }
         }
 
-        checkWriteEnrollmentAccess( reporter, enrollment, program, enrollment.getTrackedEntity(),
+        checkWriteEnrollmentAccess( reporter, enrollment, program, context,
             ownerOrgUnit );
     }
 
+    private void checkEnrollmentOrgUnit( Enrollment enrollment, TrackerImportValidationContext context,
+        TrackerImportStrategy strategy, Program program, ValidationErrorReporter reporter )
+    {
+        OrganisationUnit enrollmentOrgUnit = null;
+
+        if ( strategy.isUpdateOrDelete() )
+        {
+            enrollmentOrgUnit = context.getProgramInstance( enrollment.getEnrollment() )
+                .getOrganisationUnit();
+
+            if ( enrollmentOrgUnit == null )
+            {
+                log.warn( "ProgramInstance " + enrollment.getEnrollment()
```

Comment:
just to understand, in case we update or delete an enrollment, we allow it if the enrollment has no orgUnit assigned?
would there be another way to get the orgUnit? @enricocolasante 
I mean could someone not in the orgUnit of the enrollment delete it? I know the enrollment has none set in this case but maybe its program?

Reply:
I think this is temporary and part of a bigger problem.
OrgUnit shouldn't be null but we don't have this constraint in the DB, so some enrollments could not have and orgUnit.

Said so, I am not sure what are the rules to have the permission to delete. So I have no idea if it would be the same using the program orgUnit instead.
Maybe @stian-sandvold or @Bekkalizer have a better understanding of it

Reply:
sorry, i am missing something here. What do you mean by program orgunit? afaik program has sharing, program instance has an orgunit assigned that, for now, can be null but there is a discussion to make it not null. Does a program itself has an orgunit assigned ?

Reply:
I was just wondering if there is another entity with an orgUnit that is connected to the enrollment where the orgUnit is not null 😅 So we could use that (like a proxy) to check if the user and this orgUnit match. Program has a `Set<OrgUnits>` but it's not mapped in the programs `PreheatMapper`.

This just came to mind, since I also don't know what the risks are of someone outside the "enrollments orgUnit" (it must belong to some orgUnit I assume) deleting the enrollment are. @Bekkalizer and @stian-sandvold probably know this 😁

Reply:
just asked in the standup. @stian-sandvold clarified it for me so I'll resolve this 😄